docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
+#     problem. This leads to a faster solution. Try other solvers to see how
+#     they behave!
 
-# First we need to define a model object: default likelihood is poisson, default non-linearity is exp
-# ignore the regularizer for now, more about it later
 model = nmo.glm.GLM(regularizer=nmo.regularizer.UnRegularized(solver_name="LBFGS"))
 
-# set the weights and intercept (usually learned, but hard-coded for the example)
-# in nemos weights are (num_neurons, num_features)
-model.coef_ = np.atleast_2d(w)
-# and intercepts (num_neurons,)
-model.intercept_ = np.atleast_1d(c)
-
-# equivalently in a single step, call the `predict` method passing the current
-predicted_rate_nmo = model.predict(input_feature.d)
-predicted_rate_nmo = nap.Tsd(t=input_feature.t, d=np.squeeze(np.asarray(predicted_rate_nmo)))
-
-# %%
-# let's plot each step for 500ms
-interval = nap.IntervalSet(start=471, end=471.5)
-plt.figure(figsize=(10, 3.5))
-ax = plt.subplot(1, 3, 1)
-ax.set_title("Current $i(t)$")
-ax.plot(np.squeeze(input_feature.restrict(interval)), color="grey")
-ax = plt.subplot(1, 3, 2)
-ax.set_title("Linearly Transformed $L(t)$")
-ax.plot(np.squeeze(L.restrict(interval)), color="orange")
-ax.set_xlabel("Time (s)")
-ax = plt.subplot(1, 3, 3)
-ax.set_title(r"Rate $\lambda(t)$")
-ax.plot(np.squeeze(predicted_rate.restrict(interval)), color="tomato", label=r"$\exp(L(t))$")
-ax.plot(np.squeeze(predicted_rate_nmo.restrict(interval)), "--", label="nemos")
-plt.legend()
-plt.tight_layout()
-
-
 # %%
-# !!! info
-#     Only the weights $w$ and the intercept $c$ are learned, the non-linearity is kept fixed.
-#     In nemos, we default to the exponential, other choices such as soft-plus, are allowed. These
-#     choices guarantee convexity, i.e. a single optimal solution.
-#     In principle, one could choose a more complex non-linearity, but convexity is not
-#     guaranteed in general.
+#
+# Now that we've initialized our model with the optimization parameters, we can
+# fit our data! In the previous section, we prepared our model matrix
+# (`predictor`) and target data (`count`), so to fit the model we just need to
+# pass them to the model:
+
+model.fit(predictor, count)
 
 # %%
-# ### 3. The Poisson log-likelihood
-# The last component of the model is the poisson log-likelihood that quantifies how likely it is
-# to observe certain counts for a given firing rate time series.
-# The if $y(t)$ are the spike counts, the equation for the log-likelihood is
-# $$ \sum\_t \log P(y(t) | \lambda(t)) = \sum\_t  y(t) \log(\lambda(t)) - \lambda(t) - \log (y(t)!)\tag{2}$$
-# In nemos, the log-likelihood can be computed by calling the score method passing the predictors and the counts.
-# The method first compute the rate $\lambda(t)$ using (1) and then the likelihood using (2).
+#
+# Now that we've fit our data, we can retrieve the resulting parameters, the
+# analogues of the `weights` and `intercepts` variables used earlier in this
```

Comment:
This part seems to refer to the piece of the tutorial that got moved elsewhere?

Reply:
whoops ,yes, good catch